fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -5917,6 +5931,12 @@ class FileBackupAgentImpl {
 			printf("Restoring backup to version: %lld\n", (long long)targetVersion);
 		}
 
+		// Preload applyMutationsKeyVersionMap for only-apply-mutation-log restore. It's
+		// a workaround for case when the map is large and may exceed transaction limit.
+		if (onlyApplyMutationLogs) {
+			wait(preloadApplyMutationsKeyVersionMap(cx, randomUid, ranges, beginVersions));
```

Comment:
Why is this being done only now? I feel like it should have been an issue before these changes as well?

Reply:
Previously we only allow single begin version for a restore.. So the applyMutationsKeyVersionMap should have only one element(all keys with single begin version) so it doesn't exceed transaction commit limit. Now we want to specify a begin version for each key range, so the map becomes huge..

Reply:
I see but there's no workloads currently testing multiple begin version so this is to preemptively fix this issue?

Reply:
it'll be tested by a new workload. I'm preparing PR for it. I would keep this PR small enough for ease of code review. This PR makes sure all existing workloads that depend on existing API are not broken.